src/sklearn_raster/utils/features.py

```diff
@@ -15,3 +15,49 @@ def get_minimum_precise_numeric_dtype(value: int | float) -> np.dtype:
         if np.issubdtype(type(value), np.integer)
         else np.dtype(type(value))
     )
+
+
+def can_cast_value(value: float | int, to_dtype: np.dtype) -> bool:
+    """
+    Test whether a given numeric value can be safely cast to the target dtype.
+
+    Compared to `np.can_cast` and `np.min_scalar_type`, this check is determined based
```

Comment:
Possibly "As opposed to" or "In contrast to" rather than "Compared to".  I don't feel strongly about this though.

Reply:
Looks great!